tests/scripts/thread-cert/border_router/test_advertising_proxy.py

```diff
@@ -77,14 +77,17 @@ def test(self):
         server = self.nodes[BR]
         client = self.nodes[ROUTER]
 
+        server.srp_server_set_enabled(False)
         host.start(start_radvd=False)
         self.simulator.go(5)
 
+        self.assertEqual(server.srp_server_get_state(), 'disabled')
         server.srp_server_set_enabled(True)
         server.srp_server_set_lease_range(LEASE, LEASE, KEY_LEASE, KEY_LEASE)
         server.start()
-        self.simulator.go(5)
+        self.simulator.go(10)
         self.assertEqual('leader', server.get_state())
+        self.assertEqual(server.srp_server_get_state(), 'running')
```

Comment:
https://github.com/openthread/openthread/blob/main/tests/scripts/thread-cert/test_netdata_publisher.py might be a good place to test the `state` where all states, including `stopped`,  could be covered.